index.js

```diff
@@ -76,7 +77,62 @@ exports.wrapAsyncMethod = function (obj, method, optionsArg) {
   options.responsePath = optionsArg.responsePath;
   options.sinon = optionsArg.sinon;
 
-  const wrapper = function () {
+  if (options.isPromise) {
```

Comment:
Branching here adds many lines, most of which duplicate the rest of this method below.  This can be restructured to detect promises, and not duplicate the calls to fs.readFile and fs.writeFileSync